fbw-common/src/wasm/systems/systems/src/shared/logic_nodes.rs

```diff
@@ -0,0 +1,624 @@
+    }
+
+    pub fn update(&mut self, hi: bool, delta: Duration) -> bool {
+        let condition_met = hi == self.rising_edge;
+        if condition_met {
+            self.condition_since += delta;
+            if self.condition_since >= self.time_delay {
+                self.output = hi;
+            }
+        } else {
+            self.condition_since = Duration::ZERO;
+            self.output = hi;
+        }
+        self.output
+    }
+
+    pub fn get_output(&self) -> bool {
+        self.output
+    }
+}
+
+/// A monostable trigger, which outputs lo until it detects a rising or falling edge. At that point
+/// it will start outputting hi until the time delay period has elapsed. If the node is
+/// retriggerable, a matching edge will reset the timer. Otherwise they are ignored until
+/// the period has elapsed and the node outputs lo again.
+pub struct MonostableTriggerNode {
+    rising_edge: bool,
+    time_delay: Duration,
+    retriggerable: bool,
+    remaining_trigger: Duration,
+    last_hi: Option<bool>,
+    output: bool,
+}
+
+impl MonostableTriggerNode {
+    pub fn new(rising_edge: bool, time_delay: Duration) -> Self {
+        Self {
+            rising_edge,
+            time_delay,
+            retriggerable: false,
+            remaining_trigger: Duration::ZERO,
+            last_hi: None,
+            output: false,
+        }
+    }
+
+    pub fn new_retriggerable(rising_edge: bool, time_delay: Duration) -> Self {
+        Self {
+            rising_edge,
+            time_delay,
+            retriggerable: true,
+            remaining_trigger: Duration::ZERO,
+            last_hi: None,
+            output: false,
+        }
+    }
+
+    pub fn new_rising(time_delay: Duration) -> Self {
+        Self::new(true, time_delay)
+    }
+
+    pub fn new_falling(time_delay: Duration) -> Self {
+        Self::new(false, time_delay)
+    }
+
+    pub fn update(&mut self, hi: bool, delta: Duration) -> bool {
+        self.remaining_trigger = self
+            .remaining_trigger
+            .checked_sub(delta)
+            .unwrap_or_default();
+        if self.retriggerable || self.remaining_trigger == Duration::ZERO {
+            let condition_met =
+                self.last_hi.unwrap_or(!self.rising_edge) != hi && hi == self.rising_edge;
+            if condition_met {
+                self.remaining_trigger = self.time_delay;
+            }
+        }
+        self.last_hi = Some(hi);
+        self.output = self.remaining_trigger > Duration::ZERO;
+        self.output
+    }
+
+    pub fn output(&self) -> bool {
+        self.output
+    }
+}
+
+/// A node that detects a rising or a falling edge and will trigger exactly once. Similar to a
+/// monostable trigger node, except that the signal will immediately return, and T approaches 0.
+pub struct PulseNode {
+    rising_edge: bool,
+    last_hi: Option<bool>,
+    output: bool,
+}
+
+impl PulseNode {
+    pub fn new(rising_edge: bool) -> Self {
```

Comment:
```suggestion
    fn new(rising_edge: bool) -> Self {
```
Also here: Maybe only `new_rising` and `new_falling` should be the public "constructors"?